### PR TITLE
feat: codegen script

### LINKED ISSUE #482: chore: add codegen to Makefile
### Issue description
Add deepcopy code generation to Makefile.

### PR DESCRIPTION
**Pull request by lingsamuel**
Signed-off-by: Ling Samuel <lingsamuelgrace@gmail.com>

Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
Fixes #482
___
### New feature or improvement
- Describe the details and related test reports.

Add codegen script. Change boilerplate to Apache License.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,3 +36,5 @@ coverage.txt
 test/e2e/coverage.txt
 apisix-ingress-controller
 .actions/openwhisk-utilities
+
+.generated
```
Comment (Firstsawyou) [inline .gitignore:40]: A blank line is missing at the end of the file.

File: utils/generate-groups.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+if [ "${GENS}" = "all" ] || grep -qw "informer" <<<"${GENS}"; then
+  echo "Generating informers for ${GROUPS_WITH_VERSIONS} at ${OUTPUT_PKG}/informers"
+  "${gobin}/informer-gen" \
+           --input-dirs "$(codegen::join , "${FQ_APIS[@]}")" \
+           --versioned-clientset-package "${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-clientset}/${CLIENTSET_NAME_VERSIONED:-versioned}" \
+           --listers-package "${OUTPUT_PKG}/listers" \
+           --output-package "${OUTPUT_PKG}/informers" \
+           "$@"
+fi
```
Comment (Firstsawyou) [inline utils/generate-groups.sh:92]: Ditto.

File: utils/generate-groups.sh [hunk 3]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
```
Comment (tokers) [inline utils/generate-groups.sh:1]: Looks like this script was copied from https://github.com/kubernetes/code-generator/blob/master/generate-groups.sh.

If so, please modify the LICENSE file and claim the copyright.

Comment (tokers) [reply utils/generate-groups.sh:1]: See the bottom of https://github.com/apache/apisix-dashboard/blob/master/LICENSE for more details.

Comment (lingsamuel) [reply utils/generate-groups.sh:1]: :joy: it was kubernetes license, but it can't pass CI license-check

Comment (tokers) [reply utils/generate-groups.sh:1]: It's also Apache License but coprighted to Kubernetes Authors. @dmsolr Could you help to help us to adjust this?

**Review by tokers — APPROVED** [review]

**Review by gxthrj — APPROVED** [review]

**Pull request merged**